spec/01-unit/04-prefix_handler_spec.lua

```diff
@@ -864,6 +865,18 @@ describe("NGINX conf compiler", function()
           ngx_cfg({ wasm = true, nginx_wasm_shm_cache="10m", nginx_wasm_shm_counters="10m"}, debug)
         )
       end)
+      it("injects default configurations if wasm=on", function()
+        assert.matches(
+          ".+wasm_lua_resolver on;.+",
+          kong_ngx_cfg({ wasm = true, }, debug)
+        )
+      end)
+      it("does not inject default configurations if wasm=off", function()
+        assert.not_matches(
+          ".+wasm_lua_resolver on;.+",
+          kong_ngx_cfg({ wasm = false, }, debug)
+        )
+      end)
```

Comment:
Thank you for adding this :heart:

Tiniest of nits: can we match against the whole directive in the test? This makes the test stronger and also more friendly to git-grep:

```suggestion
      it("injects default configurations if wasm=on", function()
        assert.matches(
          ".+proxy_wasm_lua_resolver on;.+",
          kong_ngx_cfg({ wasm = true, }, debug)
        )
      end)
      it("does not inject default configurations if wasm=off", function()
        assert.not_matches(
          ".+proxy_wasm_lua_resolver on;.+",
          kong_ngx_cfg({ wasm = false, }, debug)
        )
      end)
```

Reply:
Also, one more test case to support:

```lua
      it("permits overriding proxy_wasm_lua_resolver", function()
        assert.matches(
          ".+proxy_wasm_lua_resolver off;.+",
          kong_ngx_cfg({ wasm = true,
                         -- or should this be `false`? IDK
                         nginx_http_proxy_wasm_lua_resolver = "off",
                       }, debug)
        )
      end)
```

Reply:
Good call, added!